app/publish/tus.go

```diff
@@ -200,3 +202,35 @@ func (h TusHandler) lockUpload(id string) (tusd.Lock, error) {
 	}
 	return lock, nil
 }
+
+// preCreateHook validates user access request to publish handler before we
+// attempt to start the upload procedures.
+//
+// Note that usually this should be done as part of http middleware, but TUS
+// handlers overwrite the context with context background to avoid context
+// cancellation, and so any attempt to read values from request context won't
+// work here, until they can safely pass request context to TUS handler we need
+// to decouple before and after middleware to TUS hook callback functions.
+//
+// see: https://github.com/tus/tusd/pull/342
+func (h *TusHandler) preCreateHook(hook tusd.HookEvent) error {
+	log := h.logger.WithFields(
+		logrus.Fields{
+			"module":         module,
+			"method_handler": "preCreateHook",
+		},
+	)
+
+	r := hook.HTTPRequest
+	user, err := userFromRequest(h.authProvider, r.Header, r.RemoteAddr)
+	if err != nil {
+		log.WithError(err).Error("error authenticating user")
```

Comment:
`Error` is too high a level for this message and the one below. We still want it logged in production but it's not going to be something we can act upon. `Info` would be enough.